arduino-ide-extension/src/browser/contributions/check-for-ide-updates.ts

```diff
@@ -64,6 +80,44 @@ export class CheckForIDEUpdates extends Contribution {
             e.message
           )
         );
+      })
+      .finally(() => {
+        this.setCurrentIDEVersion();
       });
   }
+
+  private async setCurrentIDEVersion(): Promise<void> {
+    try {
+      const { appVersion } = await this.appService.info();
+      const currSemVer = new SemVer(appVersion ?? '');
+      this.localStorage.setData(LAST_USED_IDE_VERSION, currSemVer.format());
+    } catch {
+      // ignore invalid versions
+    }
+  }
+
+  /**
+   * Check if user is running a new IDE version for the first time.
+   * @returns true if the current IDE version is greater than the last used version
+   * and both are non-prerelease versions.
+   */
+  private async isNewStableVersion(): Promise<boolean> {
+    try {
+      const { appVersion } = await this.appService.info();
+      const prevVersion = await this.localStorage.getData<string>(
+        LAST_USED_IDE_VERSION
+      );
+
+      const prevSemVer = new SemVer(prevVersion ?? '');
+      const currSemVer = new SemVer(appVersion ?? '');
+
+      if (prevSemVer.prerelease || currSemVer.prerelease) {
```

Comment:
```suggestion
      if (
        prevSemVer.prerelease.length > 0 ||
        currSemVer.prerelease.length > 0
      ) {
```

The `prerelease` field will be `[]` when the version is not a prerelease, and `[]` is truthy.